docs/src/tutorials/pbt-and-tla.md

```diff
@@ -0,0 +1,1001 @@
+ 
+Note that the model suggests that the action is selected before parameters (if any) are instantiated. 
+In other words, the framework does _not_ uniformly select from action instances. 
+This means that committing an approval is selected with a probability of \\(\frac{1}{6}\\) at each step, whereas \\(\text{approve}(u_1,u_2,v)\\) is selected with a probability of \\(\frac{1}{6  \cdot N_{addr}^2  \cdot N_{val}}\\), for any concrete choice of \\(u_1, u_2, v\\) (and a probability of \\(\frac{1}{6}\\) for _some_ choice of \\(u_1,u_2,v\\)).
+
+Under these constraints, what is then the probability of executing a trace, which leads to an invariant violation?
+
+Let us define the following events:
+  - \\(\omega\\): a sequence of action instances is chosen, such that the invariant is violated
+  - \\(\alpha\\): a sequence of actions is chosen, such that the invariant is violated for some selection of instance parameters
+
+Observe that \\(\alpha \cap \omega = \omega\\).
+
+We break the problem into two parts: Finding an action order, and finding parameters, assuming the order is fixed.
+Concretely, we are going to find \\(P(\alpha)\\) and \\(P(\omega \mid \alpha)\\). Then, by the conditional probability rule,
+\\[
+P(\omega) =  P(\alpha \cap \omega) = P(\omega \mid \alpha) \cdot P(\alpha)
+\\]
+
+### 8.1 Action orders that may lead to an invariant violation
+
+In this section, we determine \\(P(\alpha)\\).
+
+To find an invariant violation, we need the following five actions
+  - \\(H, L\\) (both \\(\text{submit\_approve}\\), but distinguished based on which of them is referenced by the commit action, i.e. whether the value is high or low)
+  - \\(T\\) (\\(\text{submit_transferFrom\}\\)) 
+  - \\(C_A\\) (\\(\text{commit\_approve}\\) referencing \\(H\\))
+  - \\(C_T\\) (\\(\text{commit\_transferFrom}\\))
+
+under the following constraints:
+  - \\(H\\) precedes \\(C\\) in the order
+  - \\(C_T\\) is last in the sequence of 5 actions
+
+There are 12 valid sequences that satisfy the constraints:
+\\[
+HLTC_AC_T,
+HTLC_AC_T,
+THLC_AC_T,
+LHTC_AC_T
+\\]
+\\[
+LTHC_AC_T,
+TLHC_AC_T,
+HLC_ATC_T,
+HTC_ALC_T
+\\]
+\\[
+THC_ALC_T,
+LHC_ATC_T,
+HC_ALTC_T,
+HC_ATLC_T
+\\]
+
+This means that \\(P(\alpha)\\), the probability of finding the 5 transaction sorts in the order needed to violate the invariant is \\(\frac{12}{6^5}\\).
+
+### 8.2 Parameter instantiations that may lead to an invariant violation
+
+In this section, we determine \\(P(\omega \mid \alpha )\\).
+
+Assume now that we have selected an action order, which may lead to an invariant violation. We need to instantiate the parameters to \\(H,L,T\\):
+  - \\(H: \text{submit\_approve}(sender_H, spender_H, v_H)\\)
+  - \\(L: \text{submit\_approve}(sender_L, spender_L, v_L)\\)
+  - \\(T: \text{submit\_transferFrom}(sender_T, from, to, v_T)\\)
```

Comment:
The rendering of `from` throughout the document is weird, it should be `\mathit{from}`.

Reply:
Not sure about `mathit`, but I originally had `mathrm`, which didn't work with `\_` in this rendering plugin.

Reply:
I don't think you combine `from` and `\_` anywhere?
We don't need to fix all multi-letter identifiers, but the spacing around `f` and `r` in the italicized _from_ makes it difficult to read on my machine.

Reply:
Oh, I thought you meant the from in `transferFrom`. I'm happy to make all the names single characters with indices, it will probably render much nicer that way.

Reply:
I think typing out _sender_ and _spender_ makes the document more readable, I would keep that as-is.
This only affects a few instances of _from_ and a single `_fromAddr_`, I think it's enough to wrap them in `mathrm`/`mathit`. I should probably just've made the commit, I can do it if you want.

Reply:
If you're on the latest commit then it'd probably be easier if you just added it.

Reply:
Done